software/ompi/v3.1/timeline-graph.php

```diff
@@ -16,13 +17,16 @@
 $ts1 = time() + (90 * 24 * 60 * 60);
 # 1 year beyond the first date (this i
 $ts2 = mktime(0, 0, 0, 6, 1, 2016);
+# 1 month before the first date:
+$series_start = new DateTime($first_date);
+$series_start->modify("-1 month");
 
 $ts = (int) max($ts1, $ts2);
 $beyond_end_date = strftime("%Y-%m-%d", $ts);
 
 //////////////////////////////////////////////////////////////////////////
 // v2.1 release series
-series("v3.1 series", "2018-01-01", $beyond_end_date, $data, $vpos, $graph);
+series("v3.1 series", $series_start->format("Y-m-d"), $beyond_end_date, $data, $vpos, $graph);
```

Comment:
Getting a little picky here, but I notice that your first date is 2017-10-01, which is the same as the branch date.  We typically make the first date / left edge be just a little *before* the first milestone so that the milestone isn't smushed up against the left edge of the graph.

Reply:
Ah.  Nifty!